src/pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index.jsx

```diff
@@ -0,0 +1,307 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Row, Col, Space, Collapse, Skeleton, Empty, Typography,
+} from 'antd';
+import _ from 'lodash';
+import { useSelector, useDispatch } from 'react-redux';
+import { Vega } from 'react-vega';
+import PropTypes from 'prop-types';
+import loadProcessingSettings from '../../../../../redux/actions/experimentSettings/processingConfig/loadProcessingSettings';
+import PlotStyling from '../../../../../components/plots/styling/PlotStyling';
+import { updatePlotConfig, loadPlotConfig } from '../../../../../redux/actions/componentConfig';
+import Header from '../../../../../components/plots/Header';
+import { generateSpec } from '../../../../../utils/plotSpecs/generateHeatmapSpec';
+import { loadGeneExpression, loadMarkerGenes, setGeneOrder } from '../../../../../redux/actions/genes';
+import { loadCellSets } from '../../../../../redux/actions/cellSets';
+import PlatformError from '../../../../../components/PlatformError';
+import Loader from '../../../../../components/Loader';
+import populateHeatmapData from '../../../../../components/plots/helpers/populateHeatmapData';
+import HeatmapControls from '../../../../../components/plots/styling/heatmap/HeatmapControls';
+
+const { Text } = Typography;
+const { Panel } = Collapse;
+
+const plotUuid = 'markerHeatmapPlotMain';
+const plotType = 'markerHeatmap';
+
+const route = {
+  path: 'marker-heatmap',
+  breadcrumbName: 'Marker-Heatmap',
+};
+const HeatmapPlot = ({ experimentId }) => {
+  const dispatch = useDispatch();
+  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+  const { expression: expressionData } = useSelector((state) => state.genes);
+  const { error, loading } = expressionData;
+  const cellSets = useSelector((state) => state.cellSets);
+  const { hierarchy, properties } = cellSets;
+  const selectedGenes = useSelector((state) => state.genes.expression.views[plotUuid]?.data) || [];
+  const [currentSearchedGenes, setCurrentSearchedGenes] = useState([]);
```

Comment:
I can see that the heatmap component in plots and tables doesn't need to have a state variable for the selected genes, but uses `selectedGenes` directly. Why in the case of maker heatmap we need to keep track of this and in the case of the heatmap plot we don't?

Reply:
Because in the heatmap component we use config.selectedGenes which depends on `selectedGenes`

Reply:
the heatmap in plots and tables seems to be able to do that without the need for an extra state variable. Why can't we do this in here too?